tests/js/migration-tests.js

```diff
@@ -0,0 +1,158 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+'use strict';
+
+var Realm = require('realm');
+var BaseTest = require('./base-test');
+var TestCase = require('./asserts');
+var Schemas = require('./schemas');
+
+module.exports = BaseTest.extend({
+    testMigrationFunction: function() {
+        var count = 0;
+        function migrationFunction(oldRealm, newRealm) {
+            TestCase.assertEqual(oldRealm.schemaVersion, 0);
+            TestCase.assertEqual(newRealm.schemaVersion, 1);
+            count++;
```

Comment:
As you modify the `oldRealm` inside of this function? If that would throw, then can we test that it throws (and doesn't crash) here?